### ISSUE TITLE
Inconsistent behavior using CommitNow for mutation vs. calling txn#commit()

### ISSUE THREAD

**Issue by deepakjois** [dgraph-io/dgraph#1882]
### What version of Dgraph are you using?
`master`

### Have you tried reproducing the issue with latest release?
No

### Steps to reproduce the issue (command/config used to run Dgraph).
* Start Dgraph Server.

* Run the following code using Java client and confirm that it works as expected

```java
  Transaction txn = dgraphClient.newTransaction();

    Mutation mu = Mutation.newBuilder()
            .setSetNquads(ByteString.copyFromUtf8("<_:bob> <name> \"Bob\" ."))
            .build();
    txn.mutate(mu);
    txn.commit();
    txn.discard();
``` 
* Now run a slightly modified version of the code (note the `CommitNow` set to true)

```java
  Transaction txn = dgraphClient.newTransaction();

    Mutation mu = Mutation.newBuilder()
            .setSetNquads(ByteString.copyFromUtf8("<_:bob> <name> \"Bob\" ."))
            .setCommitNow(true)
            .build();
    txn.mutate(mu);
    txn.discard();
```

This will throw an exception when calling `txn.Discard`:

```
io.grpc.StatusRuntimeException: ABORTED: Transaction has been aborted. Please retry.

	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:210)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:191)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:124)
	at io.dgraph.DgraphGrpc$DgraphBlockingStub.commitOrAbort(DgraphGrpc.java:312)
	at io.dgraph.DgraphClient$Transaction.discard(DgraphClient.java:278)
	at io.dgraph.DgraphClientTest.testDiscard(DgraphClientTest.java:168)
```

### Expected behaviour
Setting `CommitNow` or calling `txn.commit()` should not make a difference to the behavior of `txn.discard()`.

**Comment by deepakjois** [comment]
Another related issue. Setting `CommitNow()` in a mutation and then calling `Commit()` also throws an error, which is unexpected. I would  expect the subsequent call to `Commit()` to be a no-op.

```java
   Transaction txn = dgraphClient.newTransaction();

    Mutation mu = Mutation.newBuilder()
            .setSetNquads(ByteString.copyFromUtf8("<_:bob> <name> \"Bob\" ."))
            .setCommitNow(true)
            .build();
    txn.mutate(mu);
    txn.commit();
```

gives:

```

io.dgraph.TxnConflictException: Transaction has been aborted. Please retry.

	at io.dgraph.DgraphClient$Transaction.checkAndThrowException(DgraphClient.java:311)
	at io.dgraph.DgraphClient$Transaction.commit(DgraphClient.java:253)
	at io.dgraph.DgraphClientTest.testDiscard(DgraphClientTest.java:168)
```

I eyeballed the Go client code, and it is not doing anything different from the Java code. So I believe these problems would occur there as well. If there is a fix to be applied to the client, I will port it over to the Java client as well.

**Comment by peterstace** [comment]
Both of these issues are bugs in the Go client (and Java client). If a transaction has already been committed (either via Commit or via CommitNow), then Discard should be a no-op (shouldn't communicate at all with dgraph).

**Comment by peterstace** [comment]
Fixed in master at 4f7d96b
